core/src/replay_stage.rs

```diff
@@ -1227,12 +1227,15 @@ impl ReplayStage {
                 poh_slot, parent_slot, root_slot
             );
 
+            let root_distance = poh_slot - root_slot;
+
             let tpu_bank = Self::new_bank_from_parent_with_notify(
                 &parent,
                 poh_slot,
                 root_slot,
                 my_pubkey,
                 subscriptions,
+                root_distance > 500,
```

Comment:
nit: move to a constant `MAX_UNTHROTTLED_ROOT_DISTANCE`

Reply:
I was thinking about this a bit today, and was a little doubtful about how effective this would be since `root_slot` is local, so the `root_distance` is a short-lived, local concept. 

For instance, it's possible your root distance is < 500 because you're occasionally making roots, and then you switch to non-vote only mode, but then the rest of the cluster is not making roots.

I think this would be more effective if we had something like go into vote only mode if 

1. Our local root distance > 500, OR
2. > 1/3 of the vote accounts in the bank for `parent_slot` has a `root` where `root < poh_slot - 500`

This second one is a bit trickier to calculate, but I think can be done by computing the minimum root slot that has 1/3 of the stake here in `collect_vote_lockouts`. https://github.com/solana-labs/solana/blob/d0633c5547e4fd7a3099c45aedc6003397ff7db0/core/src/consensus.rs#L223. If we derive an ordered list of `(root, stake)` from smallest to largest, you can find the minimal root, `superminority_cluster_threshold_root` that `1/3` of the validators have. This can then be returned here: https://github.com/solana-labs/solana/blob/d0633c5547e4fd7a3099c45aedc6003397ff7db0/core/src/consensus.rs#L329-L335 and stored in the progress map, like the rest of the output here https://github.com/solana-labs/solana/blob/d0633c5547e4fd7a3099c45aedc6003397ff7db0/core/src/replay_stage.rs#L2184-L2200. 

We can then check that `superminority_cluster_threshold_root` for the parent slot here and see if we need to go into vote only mode

Reply:
@sakridge based on the offline chats, the `bank-only` change will be in its own PR for separate release, then I should remove its related change from this PR, should I?

Reply:
> @sakridge based on the offline chats, the `bank-only` change will be in its own PR for separate release, then I should remove its related change from this PR, should I?

merged latest v1.6, all clear